From bd20c224cf812b490b98dbf7d62aca22f0306372 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 1 Feb 2018 23:21:58 +0800
Subject: [PATCH 3346/5242] MLK-17492 drm/imx: dec400d: set read config to
 '0x0' when bypass dec400d

commit  164a61a9f47fe57086e07452a215bc2ae718c05d from
https://source.codeaurora.org/external/imx/linux-imx.git

When the DEC400D is set to bypass mode from decompressed mode,
the read config should be set to disable compression along with
the control register. Otherwise, the DEC400D cannot really leave
the decompressed mode. And the value '0x0' is suitable to be set
to read config register in this case.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dec400d.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-dec400d.c b/drivers/gpu/imx/dcss/dcss-dec400d.c
index 435cc77..639ed94 100644
--- a/drivers/gpu/imx/dcss/dcss-dec400d.c
+++ b/drivers/gpu/imx/dcss/dcss-dec400d.c
@@ -118,6 +118,8 @@ void dcss_dec400d_bypass(struct dcss_soc *dcss)
 	uint32_t control;
 	struct dcss_dec400d_priv *dec400d = dcss->dec400d_priv;
 
+	dcss_dec400d_read_config(dcss, 0, false);
+
 	control = dcss_readl(dec400d->dec400d_reg + DEC400D_CONTROL);
 	pr_debug("%s: dec400d control = %#x\n", __func__, control);
 
-- 
1.7.9.5

